packages/dds/sequence/src/test/endpointInRangeIndex.spec.ts

```diff
@@ -0,0 +1,186 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert } from "assert";
+import { v4 as uuid } from "uuid";
+import { PropertySet, Client } from "@fluidframework/merge-tree";
+import {
+	IIntervalHelpers,
+	Interval,
+	createEndpointInRangeIndex,
+	createInterval,
+} from "../intervalCollection";
+
+const reservedIntervalIdKey = "intervalId";
+
+class TestEndpointInRangeIndex {
+	private readonly intervals: Interval[];
+
+	constructor() {
+		this.intervals = [];
+	}
+
+	add(interval: Interval) {
+		this.intervals.push(interval);
+	}
+
+	findIntervalsWithEndpointInRange(start: number, end: number): Interval[] {
+		return this.intervals.filter((interval) => interval.end >= start && interval.end <= end);
+	}
+}
+
+function createTestInterval(start: number, end: number): Interval {
+	const props: PropertySet = {};
+	props[reservedIntervalIdKey] = [uuid()];
+
+	return new Interval(start, end, props);
+}
+
+function assertPlainNumberIntervals(
+	results: Interval[],
+	expectedEndpoints: { start: number; end: number }[] | Interval[],
+): void {
+	assert.equal(results.length, expectedEndpoints.length, "Mismatched result count");
+	for (let i = 0; i < results.length; ++i) {
+		assert(results[i]);
+		assert.equal(results[i].start, expectedEndpoints[i].start, "mismatched start");
+		assert.equal(results[i].end, expectedEndpoints[i].end, "mismatched end");
+	}
+}
+
+function generateRandomIntervals(count: number, rangeStart: number, rangeEnd: number) {
+	const intervals: Interval[] = [];
+
+	while (intervals.length < count) {
+		const start = getRandomNumber(rangeStart, rangeEnd);
+		const end = getRandomNumber(start, rangeEnd);
+		const interval = createTestInterval(start, end);
+		/**
+		 * Currently avoid using duplicate intervals due to the bug:
+		 * https://dev.azure.com/fluidframework/internal/_workitems/edit/4477
+		 */
+		if (
+			!intervals.some(
+				(existing) => existing.start === interval.start && existing.end === interval.end,
+			)
+		) {
+			intervals.push(interval);
+		}
+	}
+
+	return intervals;
+}
+
+function getRandomNumber(min: number, max: number): number {
+	return Math.floor(Math.random() * (max - min + 1)) + min;
```

Comment:
I prefer to make unit tests deterministic wherever possible. I'd seed the RNG source here. To do so, use `makeRandom` from `@fluid-internal/stochastic-test-utils` rather than `Math.random`.